content/en/blog/_posts/2022-03-31-Kubernetes-1-24-deprecations-and-removals

```diff
@@ -0,0 +1,85 @@
+
+As Kubernetes evolves, features and APIs are periodically revisited and removed. New features may offer an alternative or improved approach to solving an existing problem, motivating the team to remove the old method. Rather than immediately being removed, old APIs are first deprecated and then removed according to the [Kubernetes Deprecation Policy](/docs/reference/using-api/deprecation-policy/). 
+
+We want to make sure you are aware of the changes coming in the Kubernetes 1.24 release. Kubernetes 1.24 is scheduled to be released in April and includes several removed and deprecated APIs. **Removed** APIs are those that you can use in current, supported Kubernetes versions but have already been deprecated. These removals have been superseded by newer, stable/generally available (GA) APIs. **Deprecated** APIs have been marked for removal in a future Kubernetes release. 
+
+For an exhaustive list of the changes coming in Kubernetes 1.24, take a look at the [CHANGELOG](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.24.md). 
+
+## API removals and deprecations for Kubernetes 1.24
+
+### Removals
+
+The 1.24 release will stop serving the API versions listed below. 
+* [Dynamic Kubelet Configuration](https://github.com/kubernetes/enhancements/issues/281):`DynamicKubeletConfig` is used to enable the dynamic configuration of kubelet. The `DynamicKubeletConfig` flag was deprecated in Kubernetes 1.22. As of 1.24, the feature gate was removed from kubelet. See [Reconfigure kubelet](/docs/tasks/administer-cluster/reconfigure-kubelet/). Refer to the ["Dynamic kubelet config is removed" KEP](https://github.com/kubernetes/enhancements/issues/281) for more information. 
+* [Removing dockershim from kubelet](https://github.com/kubernetes/enhancements/issues/2221) (more on this below): the Container Runtime Interface (CRI) for Docker (i.e., Dockershim) is currently a built-in container runtime in the kubelet code base. It was deprecated in 1.20. As of 1.24, dockershim has been removed from kubelet. Check out this blog on [what you need to do to have a working cluster for 1.24](https://github.com/kubernetes/website/issues/30917). 
+* [VolumeSnapshot v1beta1 CRD will be removed in 1.24](https://github.com/kubernetes/enhancements/issues/177). Volume snapshot and restore functionality for Kubernetes and the [Container Storage Interface](https://github.com/container-storage-interface/spec/blob/master/spec.md) (CSI), which provides standardized APIs design (CRDs) and adds PV snapshot/restore support for CSI volume drivers, entered beta in 1.20. VolumeSnapshot v1beta1 was deprecated in 1.21. Refer to [KEP-177: CSI Snapshot](https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/177-volume-snapshot#kep-177-csi-snapshot) and [kubernetes-csi/external-snapshotter](https://github.com/kubernetes-csi/external-snapshotter/releases/tag/v4.1.0) for more information. 
```

Comment:
Is this an actual removal? What does it mean for a CRD to be “removed”? I think the change here is that VolumeSnapshot moves from *deprecated** to **unsupported**.

Once Kubernetes v1.24 becomes the _oldest supported_ version of Kubernetes (either generally, or specifically for a given CSI driver) then that driver is welcome to fully remove its support for beta versions of the VolumeSnapshot API.